File: supportPortolApp/src/app/service/uniteFabrication.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { CostumHttpResponse } from '../model/custom-http-response';
import { LigneProduction } from '../model/ligneProduction';
import { UniteFabrication } from '../model/uniteFabrication';

@Injectable({
  providedIn: 'root'
})
export class UniteFabricationService {
  
  private host  = environment.apiUrl ;

  constructor(private http : HttpClient ) {}


  public getAllUniteFab ():Observable < UniteFabrication[] > {
    return this.http.get<UniteFabrication[]>(`${this.host}/uniteFab/list`);
  }

  public addUniteFab (formData :FormData):Observable < UniteFabrication> {
    return this.http.post<UniteFabrication>(`${this.host}/uniteFab/add` , formData );
  }
  
  
  public updateUniteFab (formData :FormData):Observable <UniteFabrication > {
    return this.http.post<UniteFabrication>(`${this.host}/uniteFab/update `, formData );
  }

  public deleteUniteFab (codeUf :String ):Observable <CostumHttpResponse > {
    return this.http.delete<CostumHttpResponse>(`${this.host}/uniteFab/delete/${codeUf}`);
  }

  

  public getLignesProdsByUF (codeUf :String):Observable < LigneProduction[] > {
    return this.http.get<LigneProduction[]>(`${this.host}/uniteFab/lignesProds/${codeUf}`);
  }




  public createGroupFormData( currentCodeUf : string ,  uniteFab : UniteFabrication) : FormData {
    const formData = new FormData();
    formData.append('currentCodeUf' , currentCodeUf ) ;
    formData.append('codeUf',uniteFab.codeUf) ;
    formData.append('mapa',uniteFab.mapa) ;
    formData.append('status', JSON.stringify(uniteFab.status)) ;
    return formData ; 
  }






}

File: supportportal/src/main/java/com/supportportal/service/UniteFabricationService.java
package com.supportportal.service;

import java.io.IOException;
import java.util.List;

import com.supportportal.domain.LigneProduction;
import com.supportportal.domain.UniteFabrication;
import com.supportportal.exception.domain.CodeUniteFabExistException;
import com.supportportal.exception.domain.UniteFabNotFoundException;

public interface UniteFabricationService {

	UniteFabrication findUniteFabByCodeUf(String codeUf);

	List<UniteFabrication> getAllUniteFab();

	UniteFabrication addNewUniteFab(String codeUf, String mapa , Boolean status)throws CodeUniteFabExistException, UniteFabNotFoundException;

	UniteFabrication updateUniteFab ( String currentCodeUF, String newCodeUF, String mapa , Boolean status)throws CodeUniteFabExistException, UniteFabNotFoundException;

	void deleteUniteFab(String codeUf) throws IOException, UniteFabNotFoundException;

	List<LigneProduction> getAllLignesProdsByUF(String codeUf) throws UniteFabNotFoundException;

	void removeLigneProdFromUniteFab(String codeLp);

}
